### PR TITLE
Update Go to 1.17

### LINKED ISSUE #625: Bump Golang to 1.17
https://golang.org/doc/go1.17

### PR DESCRIPTION
**Pull request by mRrvz**
Golang has been updated to the newest release 1.17. 
Updated both local build and CI build. 
Bumped dependency packages (go.mod). 

Closes #625

### CODE REVIEW

**Comment by Totktonada** [comment]
Just to be sure: all those go.mod and go.sum changes will not break compatibility with previous Go versions, right?

**Comment by mRrvz** [comment]
@Totktonada right

**Review by Totktonada — APPROVED** [review]
I don't mind the change so.

But, please, look at the fail in CI and clarify, whether it is expected (and trackerized as a problem) or something new / unknown / requires investigation.

File: CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,10 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 
 - Ability to reverse search in ``cartridge enter`` and ``cartridge connect`` commands.
 
+### Changed
+
+- Updated Go to version 1.17
```
Comment (Totktonada) [inline CHANGELOG.md:22]: I would clarify this entry. Updated in CI, right? Whether is works on 1.17 before? Or those go.mod/sum changes were required?

**Comment by mRrvz** [comment]
> But, please, look at the fail in CI and clarify, whether it is expected (and trackerized as a problem) or something new / unknown / requires investigation.

This is a flaky test (saw him several times before), it is not related to the Golang bump. See https://github.com/tarantool/cartridge-cli/issues/519#issuecomment-911705471

File: CHANGELOG.md (hunk 1 above)
Comment (mRrvz) [reply CHANGELOG.md]: Fixed

**Review by LeonidVas — APPROVED** [review]
LGTM

File: go.mod [hunk 2]
```diff
@@ -13,30 +14,53 @@ require (
...
 	github.com/dave/jennifer v1.4.1
+	github.com/davecgh/go-spew v1.1.1 // indirect
+	github.com/docker/distribution v2.7.1+incompatible // indirect
 	github.com/docker/docker v20.10.7+incompatible
 	github.com/docker/go-connections v0.4.0 // indirect
+	github.com/docker/go-units v0.4.0 // indirect
 	github.com/fatih/color v1.7.0
 	github.com/fatih/structs v1.1.0
 	github.com/go-ole/go-ole v1.2.5 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
```
Comment (LeonidVas) [inline go.mod:25]: Funny, we are indirectly dependent on golang/protobuf and its fork)

**Pull request merged**